ang/civicase/case/details/file-tab/directives/file-filter.directive.js

```diff
@@ -54,10 +60,21 @@
     function applyTagsFilter () {
       if ($scope.customFilters.tag_id.length > 0) {
         var tagid = $scope.customFilters.tag_id;
-        $scope.fileFilter.params.tag_id = tagid.length === 1 ? tagid[0] : { IN: tagid };
+        $scope.fileFilterParams.tag_id = tagid.length === 1 ? tagid[0] : { IN: tagid };
       } else {
-        delete $scope.fileFilter.params.tag_id;
+        delete $scope.fileFilterParams.tag_id;
       }
     }
+
+    /**
+     * Call refresh function inside a timeout
+     *
+     * When `ng-change` is mentioned in `crm-entityref` directive, the change
+     * listener gets fired before the ng-model is changed.
+     * This function is created to avoid this problem
+     */
+    function refreshWithTimeout () {
+      $timeout($scope.refresh);
+    }
```

Comment:
is this issue only happening because we are using `ng-change` instead of `$watch` or is the issue happening for both anyways?

Reply:
No this is only happening for `ng-change`, and for `crm-entityref` directive only.